### PR TITLE
Attempt to roll tarball and Yosys packages for symbiflow-examples.  Name the code blocks.   Address issue #48.

### PR DESCRIPTION
**Pull request by tcal-x**
- Remove extra git clone of repo (which had mixed in code from master)
- Name all code blocks executed by tuttest
- Use "bash -ex -"

- Update conda packages and tarball to get IDELAYCTRL fix
- Remove LOC constraint on IDELAYCTRL in Linux example RTL
- Add 100T build for Linux example.

### CODE REVIEW

File: xc7/environment.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: xc7
 channels:
   - symbiflow
 dependencies:
-  - symbiflow::symbiflow-yosys=0.8_3925_g6bccd35a
-  - symbiflow::symbiflow-yosys-plugins=1.0.0.7_0032_g104f4fc
+  - symbiflow::symbiflow-yosys=0.8_6021_gd8b2d1a2
+  - symbiflow::symbiflow-yosys-plugins=1.0.0.7_0117_g160b309
```
Comment (litghost) [inline xc7/environment.yml:6]: Shouldn't this match https://github.com/SymbiFlow/symbiflow-arch-defs/blob/8b9105d2009bbd9e59282b331d34187e65891110/environment.yml#L7 ?

**Comment by tcal-x** [comment]
Error is:

```
Writing bitstream ...
Traceback (most recent call last):
  File "/opt/symbiflow/xc7/conda/envs/xc7/bin/xcfasm", line 8, in <module>
    sys.exit(main())
  File "/opt/symbiflow/xc7/conda/envs/xc7/lib/python3.8/site-packages/xc_fasm/xc_fasm.py", line 55, in main
    fasm2frames(
  File "/opt/symbiflow/xc7/conda/envs/xc7/lib/python3.8/site-packages/xc_fasm/fasm2frames.py", line 186, in fasm2frames
    assembler.parse_fasm_filename(filename_in, extra_features=extra_features)
  File "/opt/symbiflow/xc7/conda/envs/xc7/lib/python3.8/site-packages/prjxray/fasm_assembler.py", line 190, in parse_fasm_filename
    raise FasmLookupError('\n'.join(missing_features))
prjxray.fasm_assembler.FasmLookupError: Segment DB RIOI3, key RIOI3.ILOGIC_Y0.ISERDES.MODE.MASTER not found from line 'RIOI3_X43Y39.ILOGIC_Y0.ISERDES.MODE.MASTER'
Segment DB RIOI3, key RIOI3.ILOGIC_Y0.ISERDES.NETWORKING.DDR.W8 not found from line 'RIOI3_X43Y39.ILOGIC_Y0.ISERDES.NETWORKING.DDR.W8 = 1'b1'
Segment DB RIOI3, key RIOI3.ILOGIC_Y0.ISERDES.NUM_CE.N1 not found from line 'RIOI3_X43Y39.ILOGIC_Y0.ISERDES.NUM_CE.N1 = 1'b1'
Segment DB RIOI3, key RIOI3.ILOGIC_Y1.ISERDES.MODE.MASTER not found from line 'RIOI3_X43Y45.ILOGIC_Y1.ISERDES.MODE.MASTER'
etc
```

**Comment by tcal-x** [comment]
Hmm, I guess I'm encountering issue #48 , so I can't see my new test run.

**Comment by litghost** [comment]
> Hmm, I guess I'm encountering issue #48 , so I can't see my new test run.

The CI is green now, but does this comment imply we are not testing the change in CI?  I'm confused...

**Comment by tcal-x** [comment]
> > Hmm, I guess I'm encountering issue #48 , so I can't see my new test run.
> 
> The CI is green now, but does this comment imply we are not testing the change in CI? I'm confused...

Yeah, I don't fully understand the Travis flow, but it seems it clones the master repo at some point, not the candidate.   Although it's confusing, since CI **did** fail for my earlier changes, so it must use some aspects of the candidate.    Maybe it's just the tuttest execution that uses master.

I'm tempted to just ask to merge it, and watch and be ready to roll back if needed.

**Comment by litghost** [comment]
> I'm tempted to just ask to merge it, and watch and be ready to roll back if needed.

Agree.  We should definitely improve the commenting/naming in the travis.yml.  What we have now is just a bunch of "unnamed0", etc.  It is very hard to follow what is going on.

@mgielda You wrote https://github.com/antmicro/tuttest/ , but the current README for tuttest doesn't explain how to attach names to tests.  Can you please create some examples in the tuttest README for naming code blocks?  Thanks.

**Comment by litghost** [comment]
@tcal I believe I understand what Travis is doing and we should fix it up.

Changes to make:
 - Add comments to each tuttest invocation
 - Change every `bash -` to `bash -ex -`.  This enables bash command tracing and fail on error, which are both important.

I'm going to annotate what it is doing below:

There are 3 relevant tuttest invocations:

From https://github.com/SymbiFlow/symbiflow-examples/blob/6850c5785a6339b151468186d7b879118fa64c4d/.travis.yml#L20
```
tuttest README.rst unnamed0 | bash -
```

Get the latest conda installer and write to `conda_installer.sh`

From https://github.com/SymbiFlow/symbiflow-examples/blob/6850c5785a6339b151468186d7b879118fa64c4d/.travis.yml#L25
```
tuttest xc7/README.rst unnamed0 | bash -
```

Sets `INSTALL_DIR` to "/opt/symbiflow/xc7" and then installs conda to `INSTALL_DIR`, along with the packages from `environment.yml`.  This step uses the `environment.yml` that Travis checked out!  So changes to `environment.yml` are impacted here!!!  This also uses the contents to the new README, so changes to the README do affect CI!!!!!

From https://github.com/SymbiFlow/symbiflow-examples/blob/6850c5785a6339b151468186d7b879118fa64c4d/.travis.yml#L28
```
tuttest xc7/README.rst unnamed1,unnamed2,unnamed3,unnamed4,unnamed5 | bash -
```

`unnamed1` enters the conda enviroment from the previous step (good), but also checks out a copy of `master` symbiflow-examples (bad, boooo).  What we should likely do here is split the cloning of symbiflow-examples into another snippet!  There are couple of reasons to do this.  The first, and most important is that `unnamed0` already needed to have a copy of the repo cloned so it could have a `environment.yml`!!!!

**Comment by litghost** [comment]
My suggestion for the order of snippets should be:
 1. (Do not invoke in CI) Set `INSTALL_DIR`, and instruct the user to choose a writable location (e.g. ~/opt/symbiflow)
 2. (Invoke in CI) Download conda and install it to `INSTALL_DIR`.  Use conda enviroment to install git.
 3. (Do not invoke in CI) Enter conda enviroment and clone `symbiflow-examples` and cd to `symbiflow-examples`
 4. (Invoke in CI) Create new clean conda enviroment using `symbiflow-examples/environment.yml`
 5. (Invoke in CI) Build xc7 / eos examples 

@tcal-x / @mgielda Thoughts?

**Comment by tcal-x** [comment]
> @tcal I believe I understand what Travis is doing and we should fix it up.
>
>  The first, and most important is that `unnamed0` already needed to have a copy of the repo cloned so it could have a `environment.yml`!!!!

Yes -- this is wrong for the same reason for humans trying to follow the tutorial!

We should make the repo clone the first thing, and then skip that section in tuttest.

All the other things you mentioned make sense to me, and match to what I saw in the Travis log.

**Comment by pgielda** [comment]
Well tuttest is effectively following the tutorial. Which originally was just short, did require cloning only at the very end to download examples (there was no requirements.txt, all was in instructions) and repo only contained the example designs. Also it was not targetting multiple FPGAs like it is now.

I agree with the fact that now it has to be fixed -- maybe lets just not include cloning in the instructions (assume its already cloned)

**Comment by tcal-x** [comment]
Hi @litghost , does it make sense for me to merge these changes now and keep an eye on the run, then address the tuttest updates separately?

**Comment by litghost** [comment]
> Hi @litghost , does it make sense for me to merge these changes now and keep an eye on the run, then address the tuttest updates separately?

We should probably update tuttest either as part of this PR, or in a new PR seperate from this one first.

**Comment by tcal-x** [comment]
> 2. (Invoke in CI) Download conda and install it to `INSTALL_DIR`.  Use conda enviroment to install git.

Hi @litghost , I'm not sure what you mean by "Use conda environment to install git"

**Comment by litghost** [comment]
> > 1. (Invoke in CI) Download conda and install it to `INSTALL_DIR`.  Use conda enviroment to install git.
> 
> Hi @litghost , I'm not sure what you mean by "Use conda environment to install git"

I guess the question is whether we should assume if a user has git installed or not.  If we assume that the user just has to install conda, then should we use the git from conda rather than the system git?  I don't care strongly one way or another, but if we want to minimize assumptions about stuff installed on the user system, I was thinking it made some sense to use the git from conda.

**Comment by mithro** [comment]
For now, I think we can assume someone has git installed.

File: xc7/environment.yml (hunk 1 above)
Comment (tcal-x) [reply xc7/environment.yml]: Fixed.

**Review by litghost — APPROVED** [review]
LGTM
